tools/nbs/extract_images.py

```diff
@@ -19,6 +26,83 @@ def register(command):
     command.add_argument("--output", "-o", nargs="?", default=os.getcwd(), help="The folder to extract the images into")
 
 
+# Gets the image and debayers if needed, then saves the image
+def process_image(packet, output):
+    image_data = decode_image(packet.msg.data, packet.msg.format)
+
+    img = image_data[0]["image"].numpy()
+    fmt = image_data[0]["fourcc"]
+
+    # Debayer if we need to
+    if fmt == fourcc("BGGR"):
+        img = cv2.cvtColor(img, cv2.COLOR_BayerBG2RGB_VNG)
+    elif fmt == fourcc("RGGB"):
+        img = cv2.cvtColor(img, cv2.COLOR_BayerRG2RGB_VNG)
+    elif fmt == fourcc("GRBG"):
+        img = cv2.cvtColor(img, cv2.COLOR_BayerGR2RGB_VNG)
+    elif fmt == fourcc("GBRG"):
+        img = cv2.cvtColor(img, cv2.COLOR_BayerGB2RGB_VNG)
+
+    # Save the image
+    file_name = os.path.join(
+        output,
+        "{}_{:012d}.jpg".format(packet.msg.name, int(packet.msg.timestamp.seconds * 1e9 + packet.msg.timestamp.nanos)),
+    )
+    cv2.imwrite(file_name, img)
+    # Hacky! Image saved above has strange behaviour
+    Image(filename=file_name).convert("jpg").save(filename=file_name)
```

Comment:
This seems like a strange thing to do. What is wrong with the opencv saved image?

Reply:
The image it saves is somehow broken, nothing wants to read it (most things think the file is empty).

Reply:
Can you provide some examples?

Reply:
I've just tried to get an image (from the same recording) and it's now fine :shrug: